crates/guest-rust/rt/src/lib.rs

```diff
@@ -116,3 +122,55 @@ pub fn run_ctors_once() {
 /// Support for using the Component Model Async ABI
 #[cfg(feature = "async")]
 pub mod async_support;
+
+/// Cleanup helper used to deallocate blocks of canonical ABI data from
+/// lowerings.
+pub struct Cleanup {
```

Comment:
I'll note that adding this is going to wreak havoc with `cargo component`. I don't know what to do about that.

Reply:
Ok did some thinking (aka paused and did a think) -- I think we should go ahead and land this refactoring. This means that `wit-bindgen` basically can't get updated in `cargo-component` without breaking things, but it won't break anything itself. Not being able to update for a bit is I think probably the best tradeoff here for now.